### PR TITLE
Make UnexpectedConnectionClosureException public #768

### PR DESCRIPTION
**Pull request by jlprat**
Issue: #768
Make `UnexpectedConnectionClosureException` accessible for final users

### CODE REVIEW

**Comment by jlprat** [comment]
I tried to add a test to check this case, but with no luck.
What I did was:
* Start a server with own system
** Server terminates its actor system (and waits for termination) when serving its route
* Client requests route
* Expected to get the exception

I'm open to ideas on how to reproduce it.

File: akka-http-core/src/main/scala/akka/http/scaladsl/model/ErrorInfo.scala [hunk 1]
```diff
@@ -91,3 +91,6 @@ final case class EntityStreamSizeException(limit: Long, actualSize: Option[Long]
 }
 
 case class RequestTimeoutException(request: HttpRequest, message: String) extends RuntimeException(message)
+
+case class UnexpectedConnectionClosureException(outstandingResponses: Int)
```
Comment (jlprat) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/ErrorInfo.scala]: Can make this exception extend `ExceptionWithErrorInfo` as well

**Comment by jlprat** [comment]
Failed because of `sbt.ResolveException: unresolved dependency: com.typesafe.akka#akka-http_2.11;10.0.5: not found`

**Comment by jlprat** [comment]
PLS BUILD

**Comment by jlprat** [comment]
PLS BUILD

**Comment by jlprat** [comment]
PLS BUILD

**Comment by jrudolph** [comment]
Hmm, I'm actually not so sure about the underlying issue. You can handle exceptions without needing access to them. Where would be start and where would we end to expose exceptions?

**Comment by jlprat** [comment]
IMO if the exception reaches the client unwrapped, it should be public. Users might want to  recover differently according to the exception received.

Internal exceptions could still be thrown as long as they are wrapped inside a public one.

**Comment by jrudolph** [comment]
I don't think I agree. In Scala code we don't use exceptions to transport semantics so exceptions shouldn't be matched on. We could think about creating a public superclass for all of our exceptions but why would it help?

**Comment by jlprat** [comment]
Maybe we can head over the issue and discuss there if we want something else instead.

Right now the Http client `Future` is failed with that private exception, one can always recover from any exception, without caring about the type obviously. Question if that's the way we want the users to write their code.

I agree, usually exceptions are not thrown in Scala, however, this is one of the few cases where one is thrown.

**Comment by jlprat** [comment]
I guess this PR makes not much sense now (see discussion on issue: #768

**Comment by jrudolph** [comment]
Closing this one. See the discussion in #768,

**Pull request closed without merging**